src/bytes.ts

```diff
@@ -0,0 +1,265 @@
+import { assert } from './assert';
+import { add0x, assertIsHexString, remove0x } from './hex';
+
+export type Bytes = bigint | number | string | Uint8Array;
+
+/**
+ * Memoized function that returns an array to be used as a lookup table for
+ * converting bytes to hexadecimal values.
+ *
+ * The array is created lazily and then cached for future use. The benefit of
+ * this approach is that the performance of converting bytes to hex is much
+ * better than if we were to call `toString(16)` on each byte.
+ *
+ * The downside is that the array is created once and then never garbage
+ * collected. This is not a problem in practice because the array is only 256
+ * elements long.
+ *
+ * @returns A function that returns the lookup table.
+ */
+function getPrecomputedHexValuesBuilder(): () => string[] {
+  // To avoid issues with tree shaking, we need to use a function to return the
+  // array. This is because the array is only used in the `bytesToHex` function
+  // and if we were to use a global variable, the array might be removed by the
+  // tree shaker.
+  const lookupTable: string[] = [];
+
```

Comment:
can't you precompute during creation?

```typescript
function builder() {
  const lookupTable: string[] = [];
  if (lookupTable.length === 0) {
    for (let i = 0; i < 256; i++) {
      lookupTable.push(i.toString(16).padStart(2, '0'));
    }
  }

  return () => lookupTable as readonly string[];
}
```

Reply:
Yes, but I wanted to have it precomputed lazily, rather than whenever the library is imported.